include/ensmallen_bits/problems/dtlz/dtlz5_function.hpp

```diff
@@ -0,0 +1,207 @@
+      { return this -> numVariables; }
+
+      /**
+      * Set the no. of pareto points.
+      *
+      * @param numParetoPoint
+      */
+      void SetNumParetoPoint (size_t numParetoPoint)
+      { this -> numParetoPoints = numParetoPoint; }
+
+      /**
+      * Evaluate the G(x) with the given coordinate.
+      *
+      * @param coords The function coordinates.
+      * @return arma::Row<typename MatType::elem_type>
+      */
+      arma::Row<typename MatType::elem_type> g (const MatType& coords)
+      {
+        size_t k = numVariables - numObjectives + 1;
+
+        // Convenience typedef.
+        typedef typename MatType::elem_type ElemType;
+        
+        arma::Row<ElemType> innerSum(size(coords)[1], arma::fill::zeros);
+        
+        for(size_t i = numObjectives - 1;i < numVariables;i++)
+        {
+          innerSum += arma::pow((coords.row(i) - 0.5), 2); 
+        } 
+        
+        return innerSum;
+      }    
+
+      /**
+      * Evaluate the objectives with the given coordinate.
+      *
+      * @param coords The function coordinates.
+      * @return arma::Mat<typename MatType::elem_type>
+      */
+      arma::Mat<typename MatType::elem_type> Evaluate (const MatType& coords)
+      {
+        // Convenience typedef.
+        typedef typename MatType::elem_type ElemType;
+
+        arma::Mat<ElemType> objectives(numObjectives, size(coords)[1]);
+        arma::Row<ElemType> G = g(coords); 
+        arma::Row<ElemType> value = 0.5 * (1.0 + G);
+        arma::Row<ElemType> theta;
+        for(size_t i = 0;i < numObjectives - 1;i++)
+        {
+          theta = 0.5 * (1.0  + 2.0 * coords.row(i) % G) / (1.0 + G);
+          objectives.row(i) =  value %  
+                        arma::sin(theta * arma::datum::pi * 0.5);
+          value = value % arma::cos(theta * arma::datum::pi * 0.5); 
+        }
+        objectives.row(numObjectives - 1) = value;
+        return objectives;
+      }
+      
+      // Individual Objective function.
+      // Changes based on stop variable provided. 
+      struct DTLZ5Objective
+      {
+        DTLZ5Objective (size_t stop, DTLZ5& dtlz): stop(stop), dtlz(dtlz)
+        {/* Nothing to do here. */}  
+        
+        /**
+        * Evaluate one objective with the given coordinate.
+        *
+        * @param coords The function coordinates.
+        * @return arma::Col<typename MatType::elem_type>
+        */
+        typename MatType::elem_type Evaluate (const MatType& coords)
+        {
+          // Convenience typedef.
+          typedef typename MatType::elem_type ElemType;
+          ElemType value = 1.0;
+          ElemType theta;
+          ElemType G = dtlz.g(coords)[0];
+          for(size_t i = 0;i < stop;i++)
+          {
+            theta = 0.5 * (1.0  + 2.0 * coords[i] * G) / (1.0 + G); 
+            value = value * std::cos(theta * arma::datum::pi * 0.5);
+          }
+	        theta = 0.5 * (1.0  + 2.0 * coords[stop] * G) / (1.0 + G);
```

Comment:
Where is this coming from?

Reply:
I have implemented to keep updating theta with every element and multiply its cos term to for the final value according to this formula for dtlz5

![image](https://github.com/mlpack/ensmallen/assets/118750341/23e2469b-47fb-4876-a323-e3e6756e6228)

Reply:
I see thanks for the clarification.